### DISCUSSION TITLE
Improved compatibility with custom cargo.

### DISCUSSION THREAD

**Discussion by StickShiftGames** [voxeltycoon/issues#751] [Modding] [answered]
Cargo items should be made more mod-friendly. As there currently set up, any mod that adds custom cargo requires the mod to also update every vehicle's cargo file (.capacities. .stackcapacity . .heapcapacity) to add the new cargo to the Items array. And any modded vehicles need a compatibility update mod to make the modded vehicles use the new cargo. I believe there is a way to set up cargo so there compatible with all vehicles, modded or not, without the need of updating vehicles.

Instead of listing every cargo in a large array, have a single value stating the cargo type the vehicle transports, a value for each of the 3 current cargos, “boxed” (.capacities), “stack” (.stackcapacity), and “loose” (.heapcapacity) Then in the .item asset, a new value with a similar purpose is added.

Move the ‘Items’, ‘MeshUri’, ‘Rotation’, ‘Resolution’, ‘Spacing’, ‘Explosive’, and ‘ExplosionForce’ values from the .stackcapacity asset on trucks to the .item asset file or create a new file for these values. Leaving only the ‘Position’, ‘MaxWeight’, ‘LoadingSpeedMultiplier’, and ‘UnloadingSpeedMultiplier’. This way, all item-specific data in the .item file instead of having it in EVERY vehicle file.

All trucks and trains are relatively the same size or can be adjusted to be the same size. So, all trucks and cars need to list is a value stating the center of the bed, then the deals on how cargo should be loaded are gathered from the .item file, NOT the vehicles file.

With these changes, vehicles would not need to be updated to be compatible with custom cargo, all the information for ALL vehicles are in a single .item file.

**Comment by andrewpey** [comment, marked as answer]
Thank you for the valuable feedback! [Item groups](https://docs.voxeltycoon.xyz/guides/migration-guides/item-groups/) were introduced in 0.88.1 to address this.
